docs/backend/SYCL.md

```diff
@@ -291,6 +271,8 @@ For AMD GPUs we should expect at least one SYCL-HIP device [`hip:gpu`]:
 
 ### II. Build llama.cpp
 
+The SYCL backend depends on [oneMath](https://github.com/uxlfoundation/oneMath). By default it is automatically built along with the project. A specific build can be provided by setting the CMake flag `-DoneMath_DIR=/path/to/oneMath/install/lib/cmake/oneMath`.
```

Comment:
For Intel GPU, we still use Intel oneMKL included in oneAPI package now.
For non Intel GPU, maybe use pure open source oneMath.
I suggest not to impact Intel GPU in this PR, including code and document.

Reply:
One of the goal of this PR is also to unify the libraries used for the different SYCL backends. oneMath was introduced for this purpose.
We have seen issues before where a namespace used compiled when using Intel oneMKL but not oneMKL Interface (i.e. oneMath). I have introduced the `FetchContent` to avoid any impact on the user and from my benchmarks this does not affect performance either.

Reply:
SYCL backend use oneDNN to replace oneMKL to implement mul_mat() functions as default.
it's always ignored.

If don't change the related code, the oneMath/oneMKL code path won't be tested.